VisualRust/Racer/RacerSingleton.cs

```diff
@@ -67,10 +71,53 @@ private void ReinitializeRacerPaths()
             else
                 racerPathForExecution = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "Racer", BundledRacerExecutable);
             // Same for custom RUST_SRC_PATH
-            if(GetVisualRustProperty<bool>(env, nameof(RustOptionsPage.UseCustomSources)))
-                racerSourcesLocation = GetVisualRustProperty<string>(env, nameof(RustOptionsPage.CustomSourcesPath));
-            else
-                racerSourcesLocation = null;
+            var type = GetVisualRustProperty<RustOptionsPage.RustSource>(env, nameof(RustOptionsPage.SourceType));
+            switch (type) {
+                case RustOptionsPage.RustSource.EnvVariable:
+                    racerSourcesLocation = null;
+                    break;
+                case RustOptionsPage.RustSource.Custom:
+                    racerSourcesLocation = GetVisualRustProperty<string>(env, nameof(RustOptionsPage.CustomSourcesPath));
+                    break;
+                default:
+                    racerSourcesLocation = GetRustcSysroot();
+                    if (racerSourcesLocation != null && !Directory.Exists(racerSourcesLocation))
+                        racerSourcesLocation = null;
+                    break;
+            }
+
+        }
+
+        private static String GetRustcSysroot()
```

Comment:
Make all async not work =) The ICompletionSource is synchronious interface. And I cannot find any example to make it async.